src/approaches/nsrt_rl_approach.py

```diff
@@ -49,33 +65,215 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
+        # online learning cycle. The experience data is a List of (states,
+        # actions, rewards, relative_parameters) tuples. Each tuple contains
+        # information from a single instance the _Option was run. Each element
+        # of the tuple is a list: 'states' is the list of states visited,
+        # 'actions' is the list of actions taken, 'rewards' is the list of the
+        # rewards acculumated, and 'relative_params' is a list of (relative)
+        # parameter that was passed into the _LearnedNeuralParameterizedOption's
+        # regressor. The relative parameter is necessary to store because during
+        # learning, the RLOptionLearner will need to reconstruct the input to
+        # the regressor.
+        option_to_data: Dict[str,
+                             List] = {}  # option_name -> online experience
+
+        # For each training task, compute the experience data for each _Option
+        # we see used in that training task.
+        for i in range(len(self._train_tasks)):
+            plan = self._train_task_to_option_plan[i]
+            traj = self._train_task_to_online_traj[i]
+
+            curr_option_idx = 0
+            curr_option = plan[curr_option_idx]
+            parent = cast(_LearnedNeuralParameterizedOption,
+                          curr_option.parent)
+            var_to_obj = dict(
+                zip(parent._operator.parameters, curr_option.objects))  # pylint: disable=protected-access
+            var_to_unchanging_feat_ind = \
+                NSRTReinforcementLearningApproach._get_unchanging_features(
+                parent._changing_var_to_feat, var_to_obj)  # pylint: disable=protected-access
+            if curr_option.name not in option_to_data:
+                option_to_data[curr_option.name] = []
+            curr_states = [traj.states[0]]
+            curr_actions = []
+            curr_rewards = []
+            curr_relative_params = []
+            actions = (a for a in traj.actions)
+
+            for j, s in enumerate(traj.states[1:]):
+                # Store the state, action, and relative parameter.
+                curr_states.append(s)
+                curr_actions.append(next(actions))
+                curr_state_changing_feat = _create_absolute_option_param(
+                    s,
+                    parent._changing_var_to_feat,  # pylint: disable=protected-access
+                    parent._changing_var_order,  # pylint: disable=protected-access
+                    var_to_obj,
+                )
+                subgoal_state_changing_feat = curr_option.memory[
+                    "absolute_params"]
+                relative_param = subgoal_state_changing_feat - \
+                    curr_state_changing_feat
+                curr_relative_params.append(relative_param)
+
+                # In the remaining code in this loop we compute the reward.
+                reward = 0
+
+                # Add large negative reward if any object's features that are
```

Comment:
I think that we should be able to do this large negative reward computation stuff in a more concise way. for now, can we actually just take this part out of this PR, and we'll soon make another PR that is exclusively focused on this large negative reward question? my feeling is that there is already a lot of stuff going on in this PR, without this bit.

Reply:
Sure, will remove it from this PR.